### DISCUSSION TITLE
Invalid MongoDB database string

### DISCUSSION THREAD

**Discussion by hs41-18** [prisma/prisma#13764] [Q&A] [answered]
### Bug description

I am getting an error while connecting to MondoDB Atlas. 

Execution log: 

```
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db"
Error: P1013

> The provided database string is invalid. An invalid argument was provided: Database must be defined in the connection string in database URL. Please refer to the documentation in https://www.prisma.io/docs/reference/database-reference/connection-urls for constructing a correct connection string. In some cases, certain characters must be escaped. Please check the string for any illegal characters.

```

### How to reproduce

`DATABASE_URL="mongodb+srv://admin:JWFOdPcqPQBUHiCc@kudos.pxbhn.mongodb.net/?retryWrites=true&w=majority"
`

after running `npx prisma db push`

### Prisma information

```

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mongodb"
  url      = env("DATABASE_URL")
}


model User {
  id        String   @id @default(auto()) @map("_id") @db.ObjectId
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  email     String   @unique
  password  String
}
```

### Environment & setup

prisma                  : 3.15.1
@prisma/client          : 3.15.1
Current platform        : debian-openssl-1.1.x

### Prisma Version

```
3.15.1
```

**Comment by janpio** [comment, marked as answer]
The connection string is missing the database, as it says: `..udos.pxbhn.mongodb.net/?retryWrites=true&w=majority"` needs to be `...udos.pxbhn.mongodb.net/foo?retryWrites=true&w=majority"` - and you replace `foo` with your actual database name.

**Reply by hs41-18** [reply]
Hi janpio. Thanks for the quick help, much appreciated. I was confused by the connection string mentioned in Atlas (see below screenshot). Maybe worth to highlight this detail in the Prisma doc. 

<img width="762" alt="image" src="https://user-images.githubusercontent.com/9310964/173208500-64f5c4af-16b0-496a-adac-3ad152f3d7dd.png">

**Reply by janpio** [reply]
I was wondering where this came from, you are obviously not the first one who reported something like this - thanks for sharing that screenshot!

I double checked with my own Atlas code, and I can see the same - the Node.js example is indeed showing no database in the conneciton string. That is the case because the Mongo Node Client seems to prefer setting that in code: 
<img width="569" alt="image" src="https://user-images.githubusercontent.com/183673/173242445-2735cf3c-227c-489e-b1d1-a66d128e685c.png">
We'll talk to Mongo if they could add Prisma to that dropdown (although many users still will not see that) and then maybe customize our error message even more. It is not _super_ clear right now.

**Reply by GregDevLab** [reply]
Thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
